src/control/server/engine/config.go

```diff
@@ -345,68 +346,80 @@ func (c *Config) Validate() error {
 	if err := ValidateLogSubsystems(subsystems); err != nil {
 		return errors.Wrap(err, "validate engine log subsystems")
 	}
-	return c.ValidateAndAdjustPMDKEnvVar()
+	return nil
 }
 
-// Ensure proper environment variables for PMDK w/ NDCTL enabled based on
-// the actual configuration of the storage class.
-func (c *Config) ValidateAndAdjustPMDKEnvVar() error {
-	if len(c.Storage.Tiers) == 0 {
-		return nil
-	}
-
-	pmemobjConfStr, pmemobjConfErr := c.GetEnvVar("PMEMOBJ_CONF")
-	if c.Storage.Tiers[0].Class != storage.ClassDcpm {
-
-		// Disable shutdown state (SDS) (part of RAS) for RAM-based simulated SCM.
-		// RAM doesn't support this feature and trying to use
-		// it will fail the create/open operations.
-		if pmemobjConfErr != nil {
-			c.EnvVars = append(c.EnvVars, "PMEMOBJ_CONF=sds.at_create=0")
-			return nil
-		}
-
-		if !strings.Contains(pmemobjConfStr, "sds.at_create") {
-			envVars, _ := common.DeleteKeyValue(c.EnvVars, "PMEMOBJ_CONF")
-			c.EnvVars = append(envVars, "PMEMOBJ_CONF="+pmemobjConfStr+
-				";sds.at_create=0")
-			return nil
-		}
-
-		if !strings.Contains(pmemobjConfStr, "sds.at_create=0") {
-			return errors.New("env_var PMEMOBJ_CONF should contain 'sds.at_create=0' " +
-				"for non-DCPM storage class, found '" + pmemobjConfStr + "'")
-		}
-
-		return nil
-	}
-
-	// Confirm default handling of shutdown state (SDS) for DCPM storage class.
-	if pmemobjConfErr == nil && strings.Contains(pmemobjConfStr, "sds.at_create") {
-		return errors.New("env_var PMEMOBJ_CONF should NOT contain 'sds.at_create=?' " +
-			"for DCPM storage class")
-	}
-
-	// Ensure 20KiB ABT stack size for an engine with DCPM storage class.
+// Ensure at least 20KiB ABT stack size for an engine with DCPM storage class.
+func (c *Config) UpdatePMDKEnvarsStackSizeDCPM() error {
 	stackSizeStr, err := c.GetEnvVar("ABT_THREAD_STACKSIZE")
 	if err != nil {
 		c.EnvVars = append(c.EnvVars, fmt.Sprintf("ABT_THREAD_STACKSIZE=%d",
 			minABTThreadStackSizeDCPM))
 		return nil
 	}
-
+	// Ensure at least 20KiB ABT stack size for an engine with DCPM storage class.
 	stackSizeValue, err := strconv.Atoi(stackSizeStr)
 	if err != nil {
 		return errors.Errorf("env_var ABT_THREAD_STACKSIZE has invalid value: %s",
 			stackSizeStr)
 	}
-
 	if stackSizeValue < minABTThreadStackSizeDCPM {
 		return errors.Errorf("env_var ABT_THREAD_STACKSIZE should be >= %d "+
 			"for DCPM storage class, found %d", minABTThreadStackSizeDCPM,
 			stackSizeValue)
 	}
+	return nil
+}
+
+// Ensure proper configuration of shutdown (SDS) state
+func (c *Config) UpdatePMDKEnvarsPMemobjConf(isDCPM bool) error {
+	pmemobjConfStr, pmemobjConfErr := c.GetEnvVar("PMEMOBJ_CONF")
+	//also work for empty string
+	isSdsAtCreateNon := !strings.Contains(pmemobjConfStr, "sds.at_create")
```

Comment:
s/isSdsAtCreateNon/hasSdsAtCreate/ and remove inversion would I think improve clarity

Reply:
Done